paasta_tools/setup_marathon_job.py

```diff
@@ -417,9 +418,16 @@ def log_deploy_error(errormsg, level='event'):
     )
 
     if new_app_running:
+        hosts_tasks_running_on = [task.host for task in new_app.tasks]
+        draining_hosts = get_draining_hosts()
+        num_at_risk_tasks = 0
+        for host in hosts_tasks_running_on:
+            if host in draining_hosts:
+                num_at_risk_tasks += 1
+        log.debug("%s has %d tasks running on at-risk hosts." % (new_app.id, num_at_risk_tasks))
         protected_draining_tasks = set()
-        if new_app.instances < config['instances']:
-            client.scale_app(app_id=new_app.id, instances=config['instances'], force=True)
+        if new_app.instances < config['instances'] + num_at_risk_tasks:
+            client.scale_app(app_id=new_app.id, instances=config['instances'] + num_at_risk_tasks, force=True)
         elif new_app.instances > config['instances']:
             num_tasks_to_scale = max(min(len(new_app.tasks), new_app.instances) - config['instances'], 0)
             task_dict = get_old_happy_unhappy_draining_tasks_for_app(
```

Comment:
I think you need to add some code in the next few blocks of lines (433-456). This is the code that decides which of the tasks on the new app are going to be sent to the bounce function as "old" (e.g. eligible for killing).

We need to prefer tasks that are on at-risk hosts before other healthy tasks. Currently the order is draining > unhappy > happy; I think it needs to be draining > unhappy > at-risk > happy. Put them in `old_app_live_happy_tasks[new_app.id]`.